borg/repository.py

```diff
@@ -591,6 +602,28 @@ def get_fd(self, segment):
             self.fds[segment] = fd
             return fd
 
+    def _sync_and_close(self, fd):
+        fd.flush()
+        os.fdatasync(fd.fileno())
```

Comment:
this used to be .fsync previously. i read some docs about the difference and .fdatasync seems to be fine for us.

but: i also read enough to think that the sync stuff is a similar mess like the fadvise stuff, so a conservative approach would be to not change this call in this PR, but to open a ticket "fix sync for everybody" - likely needing different changes for linux, osx, windows.